Assets/_BForBoss/_Core/Scripts/ReloadViewBehaviour.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using Perigon.Weapons;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace BForBoss
+{
+    public class ReloadViewBehaviour : MonoBehaviour
+    {
+        private Image _reloadView = null;
+        private IEquipmentData _equipmentData = null;
+
+        public void Initialize(IEquipmentData equipmentData)
+        {
+            _equipmentData = equipmentData;
+        }
+
+        private void Awake()
+        {
+            _reloadView = GetComponentInChildren<Image>();
+        }
+
+        private void Update()
+        {
+            _reloadView.fillAmount = GetElapsedDurationPercentageDone();
+        }
+
+        private float GetElapsedDurationPercentageDone()
+        {
+            if (_equipmentData != null)
```

Comment:
do you want to add a check to make sure that `_equipmentData.MaxReloadDuration` cannot be 0?